software/contrib/pams.md

```diff
@@ -0,0 +1,205 @@
+# Pam's "EuroPi" Workout
+
+This program is an homage to ALM's Pamela's "NEW" Workout and Pamela's "PRO"
+Workout modules, designed for the EuroPi. It is intended to be used as a
+main clock generator, euclidean rhythm generator, clocked LFO, clocked
+random voltage source, etc... with optional quantization.
+
+The module itself will generate the master clock signal with a configurable
+BPM (1-300 BPM supported). Each output has an independently controlled
+clock multiplier or divider, chosen from the following:
+
+```
+[x16, x12, x8, x6, x4, x3, x2, x1, /2, /3, /4, /6, /8, /12, /16]
+```
+
+## I/O Mapping
+
+| I/O           | Usage
+|---------------|-------------------------------------------------------------------|
+| `din`         | External start/stop input                                         |
+| `ain`         | Routable CV input to control other parameters                     |
+| `b1`          | Start/Stop input                                                  |
+| `b2`          | Press to enter/exit edit mode. Long-press to enter/leave sub-menu |
+| `k1`          | Scroll through the settings in the current menu                   |
+| `k2`          | Scroll through allowed values for the current setting             |
+| `cv1` - `cv6` | Output signals. Configuration is explained below                  |
+
+## Menu Navigation
+
+Rotate `k1` to scroll through the current menu. Pressing and holding `b2` for 0.5s will
+enter a sub-menu. Pressing and holding `b2` again will return to the parent menu.
```

Comment:
I'm not sure I like inverting or underlining, since if anything ever has 3 layers of menus it would be nice to have a consistent look & feel.

I'll see if I can add some sort of depth icon in one of the corners, similar to the current play/pause button in the upper-right.

I'm not sure a visual indicator of whether you're on the top-level or sub-menu would necessarily be useful though; unless you know what options are in which level knowing the level you're in doesn't actually help that much.  And if you do know what options are in which level, the visual indicator probably isn't necessary.

Reply:
Since I was making menu changes anyway I've added a label to the lower-right corner of the submenu screens. e.g. if you're in the submenu for CV2 you'll see `cv2` in the lower-right.  This should hopefully reduce the ambiguity.

Reply:
I _think_ I understand what you mean now.  Try the latest revision and see if that makes more sense to you.

Reply:
Yes, this is exactly what I was thinking of. i think that it's a lot more clear now. I might put a little more space between the two items (or maybe some kind of separator) if you have the room, but it's not critical.

Reply:
I can try to tidy up the formatting a little bit. Shouldn't be too hard to add a little bit of spacing between the two parts.

Reply:
I tweaked the spacing a little more in the latest revision. Between the inverted text and a little more room between the prefix and title I'm not sure it needs a symbol to separate them. Give it a shot and let me know what you think. I might add a little more space, but I like how it currently looks about the same width as a single space between words.